cranelift/codegen/src/isa/s390x/inst/mod.rs

```diff
@@ -0,0 +1,3429 @@
+            ..
+        } => {
+            map_def(mapper, rd);
+            map_mem(mapper, mem);
+        }
+        &mut Inst::LoadRev32 {
+            ref mut rd,
+            ref mut mem,
+            ..
+        } => {
+            map_def(mapper, rd);
+            map_mem(mapper, mem);
+        }
+        &mut Inst::LoadRev64 {
+            ref mut rd,
+            ref mut mem,
+            ..
+        } => {
+            map_def(mapper, rd);
+            map_mem(mapper, mem);
+        }
+
+        &mut Inst::Store8 {
+            ref mut rd,
+            ref mut mem,
+            ..
+        } => {
+            map_use(mapper, rd);
+            map_mem(mapper, mem);
+        }
+        &mut Inst::Store16 {
+            ref mut rd,
+            ref mut mem,
+            ..
+        } => {
+            map_use(mapper, rd);
+            map_mem(mapper, mem);
+        }
+        &mut Inst::Store32 {
+            ref mut rd,
+            ref mut mem,
+            ..
+        } => {
+            map_use(mapper, rd);
+            map_mem(mapper, mem);
+        }
+        &mut Inst::Store64 {
+            ref mut rd,
+            ref mut mem,
+            ..
+        } => {
+            map_use(mapper, rd);
+            map_mem(mapper, mem);
+        }
+        &mut Inst::StoreImm8 { ref mut mem, .. } => {
+            map_mem(mapper, mem);
+        }
+        &mut Inst::StoreImm16 { ref mut mem, .. } => {
+            map_mem(mapper, mem);
+        }
+        &mut Inst::StoreImm32SExt16 { ref mut mem, .. } => {
+            map_mem(mapper, mem);
+        }
+        &mut Inst::StoreImm64SExt16 { ref mut mem, .. } => {
+            map_mem(mapper, mem);
+        }
+        &mut Inst::StoreRev16 {
+            ref mut rd,
+            ref mut mem,
+            ..
+        } => {
+            map_use(mapper, rd);
+            map_mem(mapper, mem);
+        }
+        &mut Inst::StoreRev32 {
+            ref mut rd,
+            ref mut mem,
+            ..
+        } => {
+            map_use(mapper, rd);
+            map_mem(mapper, mem);
+        }
+        &mut Inst::StoreRev64 {
+            ref mut rd,
+            ref mut mem,
+            ..
+        } => {
+            map_use(mapper, rd);
+            map_mem(mapper, mem);
+        }
+        &mut Inst::LoadMultiple64 {
+            ref mut rt,
+            ref mut rt2,
+            ref mut addr_reg,
+            ..
+        } => {
+            // FIXME: This really accesses all registers between rt and rt2.
+            // But since the instruction is only ever used after register
+            // allocation, this does not matter currently.
+            map_def(mapper, rt);
+            map_def(mapper, rt2);
+            map_use(mapper, addr_reg);
+        }
+        &mut Inst::StoreMultiple64 {
+            ref mut rt,
+            ref mut rt2,
+            ref mut addr_reg,
+            ..
+        } => {
+            // FIXME: This really accesses all registers between rt and rt2.
```

Comment:
Here, we cannot implement the correct semantics, because the remapper will not map one contiguous range to another contiguous range.  But here this really doesn't matter -- this code will never get called since these instructions are never present before register allocation.  (The only user of these instructions is the prologue/epilogue code.)

Reply:
Ah, yes, this makes sense -- thanks.